src/app/cras/edit-cra/edit-cra.component.spec.ts

```diff
@@ -90,198 +91,206 @@ describe('EditCraComponent', () => {
-    fixture.detectChanges();
-    component.showValidationMessages();
+        fixture.detectChanges();
+        component.ngOnInit();
 
-    expect(component.consultantNameInput.errors).not.toEqual([]);
-    expect(consultantNameInputValidationDiv.nodeName).not.toBe('div');
-    expect(consultantEmailInputValidationDiv.nodeName).not.toBe('div');
-    expect(missionTitleInputValidationDiv.nodeName).not.toBe('div');
-    expect(missionFinalClientInputValidationDiv.nodeName).not.toBe('div');
-  });
+        expect(spySetTitle).toHaveBeenCalledWith('Acrabadabra - Consulter un compte rendu d\'activité');
+        expect(spyDeserialize).toHaveBeenCalledWith(testReviewToken);
+      });
 
-  it('should create the cra from the form value', () => {
-    testData.cra.timesheet = miniTimesheet;
-    component.setInputsValue(testData.cra);
+      it('should disable form inputs', () => {
+        testData.mode = 'review';
+        route.queryParams = of({ data: testReviewToken });
 
-    const spyMinifyTimesheet = spyOn(component, 'minifyTimesheet')
-            .and.returnValue(miniTimesheet);
+        const spyDeserialize = spyOn(serializer, 'deserialize')
+                .and.returnValue(testData);
 
-    fixture.detectChanges();
-    component.createCRA();
+        fixture.detectChanges();
+        component.ngOnInit();
 
-    expect(component.cra).toEqual(testData.cra);
-    expect(spyMinifyTimesheet).toHaveBeenCalled();
+        expect(component.consultantNameInput.disabled).toBeTruthy();
+        expect(component.consultantEmailInput.disabled).toBeTruthy();
+        expect(component.missionTitleInput.disabled).toBeTruthy();
+        expect(component.missionFinalClientInput.disabled).toBeTruthy();
+        expect(spyDeserialize).toHaveBeenCalledWith(testReviewToken);
+      });
+    });
   });
 
-  it('should create the token', async(() => {
-    component.cra = testData.cra;
-
-    const spySerialize = spyOn(serializer, 'serialize')
-            .and.returnValue(testEditToken);
-
-    component.creatTokens();
-
-    fixture.whenStable().then(() => {
-      expect(component.editToken).toBe(testEditToken);
-      expect(spySerialize).toHaveBeenCalled();
+  describe('when submiting', () => {
+    describe('and the form is valid', () => {
+      it('should enable modal', () => {
+        component.setInputsValue(testData.cra);
+        fixture.detectChanges();
+        component.onSubmitCRA();
+
+        expect(component.showModal).toBeTruthy();
+        expect(component.showErrorModal).not.toBeTruthy();
+      });
+      it('should create edit and review tokens', () => {
+        component.setInputsValue(testData.cra);
+        component.cra = testData.cra;
+
+        const spySerialize = spyOn(serializer, 'serialize')
+                .and.returnValue(testEditToken);
+
+        fixture.detectChanges();
+        component.onSubmitCRA();
+
+        expect(component.cra.consultant.name).toBe(testData.cra.consultant.name);
```

Comment:
What you want to check here is that you have tokens with the appropriate data in it. That is not what you are doing. Take the generated tokens, deserialize them and check that the data is correct